src/keria/utils/openapi.py

```diff
@@ -0,0 +1,385 @@
+from typing import Any, Dict, List, Union, Optional
+from dataclasses import field, make_dataclass
+from marshmallow_dataclass import class_schema
+from marshmallow import fields as mm_fields
+from keri.core import serdering
+
+
+def inferType(value: Any, key: str = ""):
+    """Infer the Python type from a value, with special handling for certain keys."""
+    if key in ("a", "A"):
+        return Any
+    if isinstance(value, str):
+        return str
+    elif isinstance(value, int):
+        return int
+    elif isinstance(value, float):
+        return float
+    elif isinstance(value, bool):
+        return bool
+    elif isinstance(value, list):
+        if len(value) > 0:
+            item_type = inferType(value[0], key)
+        else:
+            item_type = str
+        return List[item_type]
+    elif isinstance(value, dict):
+        return Dict[str, Any]
+    else:
+        return Any
+
+
+def createCustomNestedField(key: str, customType: type, isOptional: bool = True) -> tuple[tuple, mm_fields.Field]:
+    """Create custom nested field for specified type."""
+    # Check if this is a Union type
+    if hasattr(customType, '__origin__') and customType.__origin__ is Union:
+        # Handle Union types
+        union_args = customType.__args__
+        
+        if len(union_args) == 2 and str in union_args:
+            list_type = next((arg for arg in union_args if hasattr(arg, '__origin__') and arg.__origin__ is list), None)
+            if list_type and len(list_type.__args__) == 1 and list_type.__args__[0] is str:
+                class StringOrStringArrayField(mm_fields.Field):
```

Comment:
Please pull the inline class out as this is bad style. This many layers of indentation I consider to be a code smell. Put the class just above this function.